src/blob/blob_client.cpp

```diff
@@ -285,6 +285,20 @@ std::future<storage_outcome<void>> blob_client::upload_block_from_stream(const s
     return async_executor<void>::submit(m_account, request, http, m_context);
 }
 
+std::future<storage_outcome<void>> blob_client::upload_block_from_stream(const std::string &container, const std::string &blob, const std::string &blockid, std::istream &is, size_t streamlen)
```

Comment:
If this API is added for unseekable stream, does it mean upload_block_from_stream in line 271 can't deal with unseekable stream? If so, we should add logic to block unseekable stream in upload_block_from_stream in line 271.

Reply:
Yes, the API cannot deal with unseekable stream. However there is no way to determine if the stream is unseekable using standard library APIs.
REF: https://stackoverflow.com/questions/9451005/c-how-to-check-if-a-stream-iostream-is-seekable